src/rays.jl

```diff
@@ -0,0 +1,81 @@
+"""
+    isintersection(b::BBox, p::Type{3, T}, d::Type{3, T})
+    isintersection(s::BSphere, p::Type{3, T}, d::Type{3, T})
+
+Return True if ray intersects a sphere or box
+"""
+
+# will go into bounding volumes
```

Comment:
Move these functions into bounding_volumes, they should be next to the BBox and BSphere methods. Maybe we should refactor bounding_volumes into a subdirectory with separate files for BBox and BSphere